libs/payments/stripe/src/lib/accountCustomer/accountCustomer.manager.ts

```diff
@@ -0,0 +1,103 @@
+// /* This Source Code Form is subject to the terms of the Mozilla Public
+//  * License, v. 2.0. If a copy of the MPL was not distributed with this
+//  * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+import { Inject, Injectable } from '@nestjs/common';
+
+import {
+  AccountDatabase,
+  AccountDbProvider,
+} from '@fxa/shared/db/mysql/account';
+import {
+  AccountCustomerNotCreatedError,
+  AccountCustomerNotDeletedError,
+  AccountCustomerNotFoundError,
+  // AccountCustomerNotUpdatedError,
+} from './accountCustomer.error';
+import {
+  createAccountCustomer,
+  deleteAccountCustomer,
+  getAccountCustomerByUid,
+  // getUidAndEmailByStripeCustomerId,
+  // updateAccountCustomer,
+} from './accountCustomer.repository';
+import {
+  CreateAccountCustomer,
+  ResultAccountCustomer,
+  // UpdateAccountCustomer,
+} from './accountCustomer.types';
+
+@Injectable()
+export class AccountCustomerManager {
+  constructor(@Inject(AccountDbProvider) private db: AccountDatabase) {}
+
+  public async createAccountCustomer(
+    input: CreateAccountCustomer
+  ): Promise<ResultAccountCustomer> {
+    const now = Date.now();
+    try {
+      const accountCustomer = await createAccountCustomer(this.db, {
+        uid: Buffer.from(input.uid, 'hex'),
+        stripeCustomerId: input.stripeCustomerId,
+        createdAt: now,
+        updatedAt: now,
+      });
+      return {
+        ...accountCustomer,
+        uid: accountCustomer.uid.toString('hex'),
+      };
+    } catch (error) {
+      throw new AccountCustomerNotCreatedError(input, error);
+    }
+  }
+
+  public async getAccountCustomerByUid(
+    uid: string
+  ): Promise<ResultAccountCustomer> {
+    try {
+      const accountCustomer = await getAccountCustomerByUid(
+        this.db,
+        Buffer.from(uid, 'hex')
+      );
+      return {
+        ...accountCustomer,
+        uid: accountCustomer.uid.toString('hex'),
+      };
+    } catch (error) {
+      throw new AccountCustomerNotFoundError(uid, error);
+    }
+  }
+
+  // public async getUidAndEmailByStripeCustomerId(
+  //   stripeCustomerId: string
+  // ): Promise<{ uid: Buffer | null; email: string | null }> {
+  //   try {
+  //     const { uid, email } = await getUidAndEmailByStripeCustomerId(
+  //       this.db,
+  //       stripeCustomerId
+  //     );
+  //     return { uid, email };
+  //   } catch (error) {
+  //     throw new AccountCustomerNotFoundError(stripeCustomerId, error);
```

Comment:
I don't believe we would be able to pass in the `uid` if we couldn't retrieve the customer from just their `stripeCustomerId`.

Reply:
But this isn't this hypothetically (because it's commented out, so I should just calm TF down) passing the `stripeCustomerId` as `uid` and then `error` object as the `stripeCustomerId`? Seems like the constructor and properties are out of sync. Or maybe the AccoutnCustomerNotFoundError class needs to drop that `uid?: string` attribute if it's an unknown wildcard.

```
AccountCustomerNotFoundError#constructor(uid, stripeCustomerId, cause)
  throw new AccountCustomerNotFoundError(stripeCustomerId, error)
```

Reply:
Heh, it's commented out because it's still a work in progress. I had jumped ahead in creating this before finishing its counterpart and its test in the repository files to ensure it works as expected before being passed in here.

As this will throw an error, instead of creating a new custom error, I decided to try futzing with `AcountCustomerNotFoundError` (which I had initially only passed in the uid and cause). I'll pick this back up when I'm done with its counterpart. :)

Reply:
@pdehaan, I just asked and `getUidAndEmailByStripeCustomerId` will not be a part of this PR, as we would first need to coordinate with FxA. *takes out MIB neuralyzer*